modules/exploits/windows/http/desktopcentral_deserialization.rb

```diff
@@ -150,16 +165,18 @@ def execute_command(cmd, _opts = {})
   def upload_serialized_payload(serialized_payload)
     print_status('Uploading serialized payload')
 
+    target_dir =
+      "#{rand_text_alpha(8..42)}\\..\\..\\..\\webapps\\DesktopCentral\\_chart"
```

Comment:
Looks good, though this is still a standard path its a lot better than using `si` every time. Approved!

Reply:
I'm pretty sure `si` stands for Source Incite, @mr_me's company. `si` was probably an intentional IOC he added to his PoC. :-)

That said, I don't think a prefix is necessary at all due to the nature of the directory traversal. I'll be removing it shortly if all goes well.

Reply:
I'm pretty sure `si` stands for Source Incite, @stevenseeley's company. `si` was probably an intentional IOC he added to his PoC. :-)

That said, I don't think a prefix is necessary at all due to the nature of the directory traversal. I'll be removing it shortly if all goes well.

Reply:
```
                String localDirToStore = baseDir + File.separator + "mdm-logs" + File.separator + customerID
                        + File.separator + deviceName + "_" + udid;                                                             // 3
```

```
wvu@kharak:~$ realpath -m /mdm-logs/foo/barsi/../../../webapps/DesktopCentral/_chart
/webapps/DesktopCentral/_chart
wvu@kharak:~$ realpath -m /mdm-logs/foo/bar/../../../webapps/DesktopCentral/_chart
/webapps/DesktopCentral/_chart
wvu@kharak:~$
```

Reply:
```
[*] Meterpreter session 1 opened (172.16.249.1:4444 -> 172.16.249.150:50092) at 2020-04-27 20:31:59 -0500
```

Reply:
#13350